### ISSUE TITLE
Message data is copied twice before it is send over UDP

### ISSUE THREAD

**Issue by jurgen-kluft** [skywind3000/kcp#121]
Is it possible to make a version of kcp that does not deal with fragment and thus enabling a possible way of implementing zero-copy of message data?

As an example when using kcp in a 100 Mbit bandwidth environment (~ 80 Mbyte/sec) i am basically copying 80 Mbyte of data twice at the sending side and twice and the receiving side, so 80 Mbyte * 4 = 320 Mbyte moved using memcpy.
When moving to 1 GiB line kcp is going to use a lot of CPU processing power just to copy data since this will become 320 Mbyte * 10 = ~3 Gbyte.

**Comment by skywind3000** [comment]
that requires a big refactor.

100 Mbit bandwidth doesn't equal to 80 MB/s, but 12.5 MB/s. while a modern CPU is capable to copy 3-5 giga bytes per seconds.

Comparing 12.5 MB/s to 5GB/s, memcpy is definitely not the bottleneck. so there is no plan to change this right now.

**Comment by jurgen-kluft** [comment]
You are right, my mistake 👍 , 100 Mbit is indeed around 10 Mbyte/sec, 1 Gbit/sec is 100 Mbyte/sec.
